diff --git a/contrib/checkpatch/checkpatch.pl b/contrib/checkpatch/checkpatch.pl
index 3cacc122c528..6a3f91f6af5f 100755
--- a/checkpatch.pl
+++ b/checkpatch.pl
@@ -1116,7 +1116,7 @@ EOM
 		print << "EOM"
 
 NOTE: If any of the errors are false positives, please report
-      them to the maintainer, see CHECKPATCH in MAINTAINERS.
+      them to the maintainer, see https://github.com/cilium/image-tools.
 EOM
 	}
 }
@@ -2697,7 +2697,7 @@ sub process {
 
 # Check if MAINTAINERS is being updated.  If so, there's probably no need to
 # emit the "does MAINTAINERS need updating?" message on file add/move/delete
-		if ($line =~ /^\s*MAINTAINERS\s*\|/) {
+		if ($line =~ /^\s*CODEOWNERS\s*\|/) {
 			$reported_maintainer_file = 1;
 		}
 
@@ -2901,7 +2901,7 @@ sub process {
 			$is_patch = 1;
 			$reported_maintainer_file = 1;
 			WARN("FILE_PATH_CHANGES",
-			     "added, moved or deleted file(s), does MAINTAINERS need updating?\n" . $herecurr);
+			     "added, moved or deleted file(s), does CODEOWNERS need updating?\n" . $herecurr);
 		}
 
 # Check for adding new DT bindings not in schema format
